twine/commands/upload.py

```diff
@@ -257,10 +258,14 @@ def upload(dists, repository, sign, identity, username, password, comment,
         if resp.is_redirect:
             raise exc.RedirectDetected(
                 ('"{0}" attempted to redirect to "{1}" during upload.'
-                 ' Aborting...').format(config["respository"],
+                 ' Aborting...').format(config["repository"],
                                         resp.headers["location"]))
         # Otherwise, raise an HTTPError based on the status code.
-        resp.raise_for_status()
+        if resp.status_code == 400 and skip_existing:
```

Comment:
Is 400 only sent because the file already exists?  isn't there some more certain way to know what happened?